### PR TITLE
SecondQuantizedMolecule: FCIDUMP file support for input/output

### PR DESCRIPTION
**Pull request by alexfleury-sb**
FCIDUMP file support for input/output, i.e. one can generate a fermionic Hamiltonian from a FCIDUMP file, or write such a file from a `SecondQuantizedMolecule` object.

As of now, the underlying functions for handling FCIDUMP files needs the `pyscf` package to work.

### CODE REVIEW

File: tangelo/toolboxes/molecular_computation/tests/test_fcidump.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+        secmol = SecondQuantizedMolecule(mol_H2_sto3g.xyz, solver=IntegralSolverFCIDUMP(pwd_this_test + "/data/H2_sto3g.fcidump"))
+        self.assertTrue(ferm_op_ref.isclose(secmol.fermionic_hamiltonian))
+
+    def test_write_fcidump(self):
+        """Writing a FCIDUMP file from a SecondQuantizedMolecule. The test
+        checks if the fermionic operator can be generated from the newly
+        created FCIDUMP file.
+        """
+        tmp = tempfile.NamedTemporaryFile()
```
Comment (ValentinS4t1qbit) [inline tangelo/toolboxes/molecular_computation/tests/test_fcidump.py:58]: This is cool. Are we creating a tmp file that is automatically named with something unique and then cleaned up when leaving the scope ?

File: tangelo/toolboxes/molecular_computation/__init__.py [hunk 2]
```diff
@@ -13,5 +13,6 @@
 # limitations under the License.
 
 from tangelo.toolboxes.molecular_computation.integral_solver import IntegralSolver, IntegralSolverEmpty
+from tangelo.toolboxes.molecular_computation.integral_solver_fcidump import IntegralSolverFCIDUMP
```
Comment (ValentinS4t1qbit) [inline tangelo/toolboxes/molecular_computation/__init__.py:16]: Pas sur qu'on ait besoin d'une nouvelle classe. Voir autres commentaires

**Review by ValentinS4t1qbit — CHANGES_REQUESTED** [review]
I think a better use of inheritance and doing surgery on the right IntegralSolver class may reduce amount of code and facilitate maintenance.

File: tangelo/toolboxes/molecular_computation/tests/test_fcidump.py (hunk 1 above)
Comment (alexfleury-sb) [reply tangelo/toolboxes/molecular_computation/tests/test_fcidump.py:58]: Yes!

**Review by ValentinS4t1qbit — CHANGES_REQUESTED** [review]
I apologize, I started a PR and never finished it. Minor things here too

**Review by ValentinS4t1qbit — APPROVED** [review]
Great job, thanks !

**Pull request merged**